_python/main/sanitize.py

```diff
@@ -0,0 +1,140 @@
+import bleach
+from functools import lru_cache
+
+"""
+    This file handles sanitizing user-submitted HTML, stored in ContentNode.headnote and TextBlock.content.
+    During save, we pass all submitted HTML through bleach.clean(), which reformats the HTML and applies an allow-list.
+    
+    Our allow-list has two sources of truth for tags, attributes, and styles that should be preserved:
+    
+    - ones can be created by ckeditor, our frontend wysiwyg editor.
+    - "legacy" ones that already exist in legacy HTML in the browser and that we don't want to strip yet.
+    
+    The second source could be reduced as we decide that certain things don't have to be preserved.
+"""
+
+## helpers ##
+
+def get_words(s):
```

Comment:
This helper is a great idea! I definitely would have just made a list of individual strings.... This is SO much easier to read!